src/Compilers/CSharp/Portable/Emitter/EditAndContinue/EmitHelpers.cs

```diff
@@ -55,7 +55,7 @@ internal static class EmitHelpers
             {
                 // TODO: https://github.com/dotnet/roslyn/issues/9004
                 diagnostics.Add(ErrorCode.ERR_ModuleEmitFailure, NoLocation.Singleton, compilation.AssemblyName, e.Message);
-                return new EmitDifferenceResult(success: false, diagnostics: diagnostics.ToReadOnlyAndFree(), baseline: null, updatedMethods: updatedMethods.ToImmutableAndFree(), updatedTypes: updatedTypes.ToImmutableAndFree());
+                return new EmitDifferenceResult(success: false, diagnostics: diagnostics.ToReadOnlyAndFree(), baseline: null, updatedMethods: updatedMethods.ToImmutableAndFree(), changedTypes: changedTypes.ToImmutableAndFree());
```

Comment:
Looks like `changedTypes` is always empty here. Let's just use an empty instance then. The ArrayBuilder can be moved closer to usage (`SerializeToDeltaStreams` below).